index.js

```diff
@@ -26,17 +26,18 @@ var prettyMs = require('pretty-ms')
 var tablify = require('tablify').tablify
 var sum = require('math-sum')
 var ncp = denodeify(require('ncp'))
+var yargs = require('yargs')
 temp.track()
 
 function getDeps () {
   return readFile('package.json', 'utf8').then(function (str) {
     var json = JSON.parse(str)
     var deps = extend({}, json.dependencies)
-    if (process.argv.indexOf('--production') === -1) {
+    if (!argv["production"]) {
       extend(deps, json.devDependencies)
     }
     // Include optionalDependencies only if --no-optional argument is absent
-    if (process.argv.indexOf('--no-optional') === -1) {
+    if (!argv["no-optional"]) {
```

Comment:
This needs double-checking; I believe yargs has some subtleties with `no-`-prefixed options, but will investigate myself